CRM/Price/BAO/PriceField.php

```diff
@@ -814,9 +814,16 @@ public static function priceSetValidation($priceSetId, $fields, &$error, $allowN
       list($componentName) = explode(':', $fields['_qf_default']);
       // now we have all selected amount in hand.
       $totalAmount = array_sum($selectedAmounts);
+      $actualTotalAmount = CRM_Utils_Array::value('partial_payment_total', $fields, CRM_Utils_Array::value('total_amount', $fields));
       if ($totalAmount < 0) {
         $error['_qf_default'] = ts('%1 amount can not be less than zero. Please select the options accordingly.', array(1 => $componentName));
       }
+      elseif ($totalAmount > 0 &&
+        $actualTotalAmount >= $totalAmount && // if total amount is equal to all selected amount in hand
+        (CRM_Utils_Array::value('contribution_status_id', $fields) == CRM_Core_PseudoConstant::getKey('CRM_Contribute_DAO_Contribution', 'contribution_status_id', 'Partially paid'))
+      ) {
+        $error['total_amount'] = ts('For partially paid contribution, amount must be less then the sum of all selected amount in hand');
```

Comment:
ts('You have specified the status Partially Paid but have entered an amount that equals or exceeds the amount due. Please adjust the status of the payment or the amount');